### PR TITLE
refactor: extract scoping action creator behavior

### PR DESCRIPTION
**Pull request by smoralesd**
#### Description of changes

Move scoping related functionality out of the `GlobalActionCreator` and into the new `ScopingActionCreator`. Cleaning `GlobalActionCreator` so we get classes/objects that follow Single Responsibility Principle.

#### Pull request checklist

- [x] Addresses an existing issue: Fixes #0000
   - does not apply
- [x] Added relevant unit test for your changes. (`npm run test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] Ran precheckin (`npm run precheckin`)
- [x] (UI changes only) Added screenshots/GIFs to description above
   - does not apply
- [x] (UI changes only) Verified usability with NVDA/JAWS
   - does not apply

### CODE REVIEW

File: src/tests/unit/tests/background/global-action-creators/global-action-creator.test.ts [hunk 1]
```diff
@@ -217,7 +160,7 @@ class GlobalActionCreatorValidator {
         commandActions: this.commandActionsContainerMock.object,
         featureFlagActions: this.featureFlagActionsContainerMock.object,
         launchPanelStateActions: this.launchPanelStateActionsContainerMock.object,
-        scopingActions: this.scopingActionsContainerMock.object,
+        scopingActions: null,
```
Comment (waabid) [inline src/tests/unit/tests/background/global-action-creators/global-action-creator.test.ts]: Can we get rid of this?

Comment (smoralesd) [reply src/tests/unit/tests/background/global-action-creators/global-action-creator.test.ts]: Not without casting. I can go either way, let me know

File: src/tests/unit/tests/background/global-action-creators/scoping-action-creator.test.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    let testSubject: ScopingActionCreator;
+
+    beforeEach(() => {
+        interpreterMock = Mock.ofType<Interpreter>();
+        scopingActionsMock = Mock.ofType<ScopingActions>();
+
+        testSubject = new ScopingActionCreator(interpreterMock.object, scopingActionsMock.object);
+    });
+
+    it('handles GetcurrentState', () => {
```
Comment (peterdur) [inline src/tests/unit/tests/background/global-action-creators/scoping-action-creator.test.ts]: Typo: should be Get**C**urrentState

Comment (smoralesd) [reply src/tests/unit/tests/background/global-action-creators/scoping-action-creator.test.ts]: updated

**Pull request merged**